File: C/poo.c
//Aprendendo POO
#include <stdio.h>
#include <stdlib.h>

// Definindo a classe Carro
typedef struct {
    char marca[20];
    int ano;
} Carro;

// Método para imprimir os detalhes do carro
void imprimirDetalhes(Carro *carro) {
    printf("Marca: %s\n", carro->marca);
    printf("Ano: %d\n", carro->ano);
}

int main() {
    // Criando um objeto da classe Carro
    Carro *meuCarro = malloc(sizeof(Carro));
    
    // Atribuindo valores aos atributos do carro
    strcpy(meuCarro->marca, "Toyota");
    meuCarro->ano = 2020;
    
    // Imprimindo os detalhes do carro
    imprimirDetalhes(meuCarro);
    
    // Liberando a memória alocada
    free(meuCarro);
    
    return 0;
}

File: Java/poo.java
public class Carro {
    private String marca;
    private int ano;

    // Construtor da classe Carro
    public Carro(String marca, int ano) {
        this.marca = marca;
        this.ano = ano;
    }

    // Método para imprimir os detalhes do carro
    public void imprimirDetalhes() {
        System.out.println("Marca: " + marca);
        System.out.println("Ano: " + ano);
    }

    public static void main(String[] args) {
        try {
            // Criando um objeto da classe Carro
            Carro meuCarro = new Carro("Toyota", 2020);
            
            // Imprimindo os detalhes do carro
            meuCarro.imprimirDetalhes();
        } catch (Exception e) {
            System.out.println("Ocorreu um erro: " + e.getMessage());
        }
    }
}
